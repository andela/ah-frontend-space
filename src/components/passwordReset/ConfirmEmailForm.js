import React from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import Input from '../common/Input';

const ConfirmEmailForm = ({
  email, onChange, loading, onSubmit,
}) => (


  <form onSubmit={onSubmit}>
    <div className="col-md-9 col-sm-12">
      <div className="form-group form-group-lg">
        <Input
          id="confirm-email"
          type="email"
          name="email"
          onChange={onChange}
          placeholder="Enter your email here"
          value={email}
        />
        { loading
          ? (
            <Loader
              type="Puff"
              color="#00bfff"
              height="20"
              width="20"
            />
          )
          : (
            <Input
              type="submit"
              id="submit"
              disabled={loading}
              className="btn-signin btn-verify"
              value="Verify"
            />
          )}
      </div>
    </div>
  </form>
);


ConfirmEmailForm.propTypes = {
  email: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

ConfirmEmailForm.defaultProps = {
};

export default ConfirmEmailForm;
