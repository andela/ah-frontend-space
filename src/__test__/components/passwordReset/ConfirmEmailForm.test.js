import React from 'react';
import { shallow } from 'enzyme';

import ConfirmEmailForm from '../../../components/passwordReset/ConfirmEmailForm';

function renderConfirmEmailForm(args) {
  const defaultProps = {
    email: '',
    loading: false,
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    value: 'email@email.com',
  };
  const props = { ...defaultProps, ...args };
  return shallow(<ConfirmEmailForm {...props} />);
}

describe('ConfirmEmailForm', () => {
  it('should match snapshot', () => {
    const wrapper = renderConfirmEmailForm();
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const wrapper = renderConfirmEmailForm({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });
});
