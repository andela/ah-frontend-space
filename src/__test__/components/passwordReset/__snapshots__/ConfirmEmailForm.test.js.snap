// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailForm should match snapshot 1`] = `ShallowWrapper {}`;

exports[`ConfirmEmailForm should match snapshot 2`] = `ShallowWrapper {}`;
